radis/lbl/base.py

```diff
@@ -1929,28 +2055,67 @@ def calc_populations_eq(self, Tgas):
         self.profiler.start("calc_eq_population", 2)
         #            printg('> Calculating equilibrium populations')
 
-        # Load partition function values
-        dgb = df1.groupby(by=["id", "iso"])
+        Qgas = {}
 
-        # ... optimize by filling all with first isotope first
-        id1, iso1 = list(dgb.indices.keys())[0]
-        molecule1 = get_molecule(id1)
-        state = self.input.state
-        parsum1 = self.get_partition_function_interpolator(molecule1, iso1, state)
-        df1["Qgas"] = parsum1.at(Tgas)
+        if "id" in df1 and "iso" in df1:
+            # Load partition function values
+            dgb = df1.groupby(by=["id", "iso"])
+
+            # ... now fill the rest:
+            for (id, iso), idx in dgb.indices.items():
+                molecule = get_molecule(id)
+                state = self.input.state
+                parsum = self.get_partition_function_interpolator(molecule, iso, state)
+                Qgas[iso] = parsum.at(Tgas)
+
+                if radis.DEBUG_MODE:
+                    assert (df1.loc[idx, "id"] == id).all()
+                    assert (df1.loc[idx, "iso"] == iso).all()
 
-        # ... now fill the rest:
-        for (id, iso), idx in dgb.indices.items():
-            if (id, iso) == (id1, iso1):
-                continue
+        elif "id" not in df1 and "iso" in df1:
+            # Load partition function values
+            dgb = df1.groupby(by=["iso"])
+
+            # ... optimize by filling all with first isotope first
```

Comment:
This optimization doesn't seem to work anymore with the new implementation where no Qgas is created. Use your proof of concept code to test it, but I think it should be fully removed

Reply:
Tested. Removed.